component-samples/demo/README.md

```diff
@@ -216,90 +216,87 @@ To enable ServiceInfo transfer to a Device with a given GUID, follow the steps b
-
-**Step 1:** Convert the certificate and private key into 'PKCS12' format:
-
-`$ openssl pkcs12 -export -in certificate.pem -inkey private-key.pem -name newkeypair -out src-keystore.p12`
-
-**Step 2:** Delete an existing alias (only needed if there is a need to replace PrivateKeyEntry having a particular algorithm):
-
-`$ keytool -delete -alias newkeypair -keystore keystore.jks`
-
-**Step 3:** Import the above generated source PKCS12 file into the existing destination keystore file 'dest-keystore.p12' located with alias 'newkeypair'.
-
-`$ keytool -importkeystore -destkeystore path/to/dest-keystore.p12 -srckeystore src-keystore.p12 -srcstoretype PKCS12 -alias newkeypair`
-
-***NOTE***: The password entered in Step 1 to generate the src-keystore.p12 must be the same as that of dest-keystore.p12, that is, the password of the newly created keystore must match the existing keystore where it will be imported to.
-
-## Exporting an Existing Certificate from Keystore
+```
+$ bash scripts/keys_gen.sh .
+$ cp -r creds/* .
+```
 
-Assuming that there is an existing certificate and private key stored in the keystore as a PrivateKeyEntry under the alias 'newkeypair', run the following command to extract the certificate into <certificate.pem\>:
+This would generate all necessary credentails in a folder named 'creds' in current working
+directory.
 
-**Step 1:** Get the list of key-pairs, along with their respective aliases, from the keystore:
+```
+$ tree creds
+creds
+├── aio
+│   ├── certs
+│   │   ├── ssl.p12
+│   │   └── truststore
+│   ├── creds.env
+│   ├── manufacturer_keystore.p12
+│   ├── owner_keystore.p12
+│   └── resources
+│       └── owner_customer1.pem
+├── device
+│   ├── device_ec256.pem
+│   ├── device_ec384.pem
+│   └── device.pem
+├── manufacturer
+│   ├── certs
+│   │   └── ssl.p12
+│   ├── creds.env
+│   ├── manufacturer_keystore.p12
+│   ├── owner_pub_keys.pem
+│   └── reseller_pub_keys.pem
+├── owner
+│   ├── certs
+│   │   ├── ssl.p12
+│   │   └── truststore
+│   ├── creds.env
+│   ├── owner2_pub_keys.pem
+│   ├── owner_keystore.p12
+│   └── owner_pub_keys.pem
+├── reseller
+│   ├── certs
+│   │   └── ssl.p12
+│   ├── creds.env
+│   ├── owner_pub_keys.pem
+│   └── reseller_keystore.p12
+└── rv
+    ├── certs
+    │   └── ssl.p12
+    ├── config.properties
+    └── creds.env
+```
 
-`$ keytool -list -v -keystore /path/to/dest-keystore.p12`
+Run the script with '-h' flag to print the syntax.
 
-**Step 2:** Export the certificate from the keystore using any of the aliases present in the keystore**:**
+```
+$ bash scripts/keys_gen.sh -h
+```
 
-`$ keytool -exportcert -alias newkeypair -file <certificate.pem> -rfc -keystore path/to/dest-keystore.p12`
+## Customize for multi-machine setup
 
-## Removing an Existing Key-Pair from Keystore
+While testing for a scenario where different components are running in different
+machines, the SSL certificate needs to be updated to include the DNS / IP of
+these machines are SubjectAltName.
```

Comment:
is this sentence complete?

Reply:
are -> in 

Would be updated as below.

While testing for a scenario where different components are running in different
machines, the SSL certificate needs to be updated to include the DNS / IP of
these machines in its SubjectAltName.

Reply:
Updated 9a6ec2e